_posts/2019-01-28-lecture-04.md

```diff
@@ -0,0 +1,681 @@
+Initialization: If $G$ is directed, first moralize $G$.   
+Procedure: For each node $X_i$ in $I$, at each step connect all of the remaining neighbors of $X_i$ and remove $X_i$ from the graph. 
+
+**Reconstituted Graph $G’_I(V, E’)$**: also named induced graph   
+
+Note: the $I$ is an elimination ordering, for different $I$ reconstituted graph is different.
+
+Definition: the reconstituted graph $G’_I(V, E’)$ is a graph whose edge set $E’$ is a superset of $E$ containing all edges of $E$ and any new edges created during a run of **Graph Elimination Algorithm**. 
+
+Reconstituted graph records the **elimination cliques** created in graph elimination algorithm. At each step before we remove a node $X_i$ from graph, connecting all neighbors of $X_i$ created a clique, which is the **elimination cliques** 
+
+**Correspondence between intermediate terms in VE and elimination cliques in GE:**   
+
+Following the corresponding steps of VE and GE, it’s easy to see that at each elimination step, the scope of generated intermediate term in VE is just he elimination clique generated in GE. The following figure shows this relationship based on the example we introduced before:
+<figure>
+  <div class="row">
+   <div class="col three">
+     <img src="{{ '/assets/img/notes/lecture-04/elimination_clique.jpg' | relative_url }}" />
+   </div>
+  </div>
+</figure> 
+
+**Theorem:**  
+
+1. The scope of every factor generated during the variable elimination process is a clique in reconstituted graph $G’_I(V, E’)$.
+
+2. Every maximal clique in reconstituted graph $G’_I(V, E’)$ is the scope of some intermediate factor in the computation. 
+
+The proof of the theorem can be found in Chapter 9 of Koller’s PGM textbook. This theorem tells us the scope of intermediate factors which is a elimination clique, is a clique in reconstituted graph. What’s more, the scope of the largest intermediate factor is a the largest maximal clique in reconstituted graph.
+
+### Complexity Analysis in Graph Perspective
+
+In the beginning of this section we have argued that the bottleneck of VE’s complexity is determined by the scope size of the maximum intermediate factor generated in the procedure of VE. In above subsection we have shown that each intermediate factor in VE is an elimination clique in graph elimination algorithm, and the largest elimination clique is also a largest maximal clique in reconstituted graph.
+
+Then, given an elimination ordering $I$, the problem of get the bottleneck maximum scope size $N_{max}$ of intermediate factors is equivalent of finding the largest clique in reconstituted graph $G’_I(V, E’)$, which is a pure graph theoretic question. And there are efficient mature algorithm for solving this problem. 
+
+### Elimination Ordering 
+
+We can define the **width** of a reconstituted graph as the size of largest clique minus 1. Let $w_{G,I}$ be the width of $G’_I(V, E’)$. For different ordering $I$, $w_{G,I}$ is different. 
```

Comment:
```suggestion
We can define the **width** of a reconstituted graph as the size of largest clique minus 1. Let $w_{G,I}$ be the width of $G_I'(V, E’)$. For different ordering $I$, $w_{G,I}$ is different. 
```

Reply:
Not sure why is this breaking up in the website..

Reply:
We couldn't figure that out either, hopefully your suggestion helps.